src/txmempool.cpp

```diff
@@ -1105,6 +1106,96 @@ bool CTxMemPool::DumpMempool(FopenFn mockable_fopen_function, bool skip_file_com
     return true;
 }
 
+void CTxMemPool::LoadMempool(std::function<bool(const CTransactionRef&, int64_t)> simple_atmp, FopenFn mockable_fopen_function)
+{
+    _LoadMempool(simple_atmp, mockable_fopen_function);
+    SetIsLoaded(!ShutdownRequested());
```

Comment:
This is really confusing. `_LoadMempool` may return false if `ShutdownRequested()`, but that's ignored here and it's checked again.

It's also not clear _why_ ShutdownRequested() may have been called. Might it be coming from some low-level operation stemming from `_LoadMempool` itself? Or is it `simple_atmp` we're worried about?

I see that you described the confusing semantics in the commit message, but I'm not convinced that it's ok to double-down on the wonky behavior before improving it some :(

Reply:
Followup entry added in umbrella issue #24303